manifest.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Check if binary path is provided
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <path-to-binary>"
+    exit 1
+fi
+
+BINARY_PATH="$1"
+BINARY_NAME=$(basename "$BINARY_PATH")
+
+# Extract version from binary name
+VERSION=$(echo "$BINARY_NAME" | sed -E 's/viam-agent-v([0-9]+\.[0-9]+\.[0-9]+(-dev\.[0-9]+)?).*/\1/')
+
+# Determine if this is a prerelease
+if [[ "$VERSION" == *"-dev"* ]]; then
+    BINNAME="prerelease/$BINARY_NAME"
+else
+    BINNAME="$BINARY_NAME"
+fi
+
+# Set platform based on architecture in binary name
+if [[ "$BINARY_NAME" == *"x86_64"* ]]; then
```

Comment:
I say this in the tone of the trailer for a horror movie but remember, windows is coming

linux/amd64 isn't the only amd64 anymore

I think you need to do one of:
- handle `*windows-x86_64` first, and make sure `*x86_64` is an else case
- always include OS in the filename

Reply:
you're assuming I haven't created an entire parallel manifest system on my branch that will lead to terrible merge conflicts